### PR TITLE
demo: Form + Modal 优化

### PR DESCRIPTION
**Pull request by crazyair**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by crazyair** [comment]
使用 `ModalCloseDestroyChildren` 组件，达到 `Modal` 打开加载组件，`Modal` 关闭卸载组件，Form 就可以写到 `Model` 组件里

**Comment by crazyair** [comment]
@afc163 @zombieJ 

使用 `ModalCloseDestroyChildren` 组件，达到 `Modal` 打开加载组件，`Modal` 关闭卸载组件，Form 就可以写到 `Model` 组件里

File: components/form/demo/form-in-modal.tsx [hunk 1]
```diff
@@ -1,33 +1,99 @@
...
+          open,
+          onCancel: () => {
+            onCancel?.();
+            setOpen(false);
+          },
+          onDestroy: () => setLoad(undefined),
+        })}
+    </div>
+  );
+};
```
Comment (afc163) [inline components/form/demo/form-in-modal.tsx:45]: 这个封装还是比较高阶，导致这个例子的复杂度会上升，用户如果有一些自定义需求会不太改的动。

File: components/form/demo/form-in-modal.tsx [hunk 2]
```diff
@@ -1,33 +1,99 @@
...
+        <Form
+          layout="vertical"
+          form={form}
+          name="form_in_modal"
+          initialValues={initialValues}
+          onFinish={onCreate}
+        >
+          {node}
+        </Form>
+      )}
```
Comment (afc163) [inline components/form/demo/form-in-modal.tsx:89]: 使用 modalRender 倒是个不错的思路。

File: components/form/demo/form-in-modal.tsx (hunk 1 above)
Comment (crazyair) [reply components/form/demo/form-in-modal.tsx:45]: 复杂的这个组建是可复用的，我觉得还好。主要是 Modal 能支持打开加载，关闭卸载，用户使用时会变的简单很多

**Pull request closed without merging**